### PR TITLE
Add delete statement suggestions

### LINKED ISSUE #4: Support `DELETE FROM {table}` for `genericAutocompleteParser`

### PR DESCRIPTION
**Pull request by NikitaShkaruba**
Closes #4

### CODE REVIEW

File: src/parsing/parsers/generic/jison/delete/delete.jison [hunk 1]
```diff
@@ -23,3 +23,24 @@ DeleteStatement
...
+
+DataManipulation_EDIT
+ : DeleteStatement_EDIT
+ ;
+
+DeleteStatement_EDIT
+ : 'DELETE' 'CURSOR'
+   {
+     parser.suggestKeywords(['FROM']);
+   }
```
Comment (NikitaShkaruba) [inline src/parsing/parsers/generic/jison/delete/delete.jison:35]: Suggest `FROM` when your query is `DELETE |`. 
> | is a cursor

File: src/parsing/parsers/generic/jison/delete/delete.jison [hunk 2]
```diff
@@ -23,3 +23,24 @@ DeleteStatement
...
+ | 'DELETE' 'FROM' 'CURSOR'
+   {
+     parser.suggestTables();
+     parser.suggestDatabases({ appendDot: true });
+   }
```
Comment (NikitaShkaruba) [inline src/parsing/parsers/generic/jison/delete/delete.jison:40]: Suggest tables (and databases for postgres) when your query is `DELETE FROM |`

File: src/parsing/parsers/generic/jison/delete/delete.jison [hunk 3]
```diff
@@ -23,3 +23,24 @@ DeleteStatement
...
+ | 'DELETE' 'FROM' SchemaQualifiedTableIdentifier 'CURSOR'
+   {
+     parser.suggestKeywords(['WHERE']);
+   }
```
Comment (NikitaShkaruba) [inline src/parsing/parsers/generic/jison/delete/delete.jison:44]: Suggest `WHERE` when your query is `DELETE FROM table |`

File: src/parsing/parsers/generic/jison/delete/delete.jison [hunk 4]
```diff
@@ -23,3 +23,24 @@ DeleteStatement
...
+ | 'DELETE' 'FROM' SchemaQualifiedTableIdentifier WhereClause_EDIT
```
Comment (NikitaShkaruba) [inline src/parsing/parsers/generic/jison/delete/delete.jison:45]: Suggest the rest of the query if you are in the `WhereClause`

File: src/parsing/parsers/generic/jison/delete/delete.test.json [hunk 5]
```diff
@@ -19,5 +19,127 @@
...
+    "beforeCursor": "",
+    "afterCursor": "",
+    "containsKeywords": [
+      "DELETE"
+    ],
+    "noErrors": true,
+    "expectedResult": {
+      "lowerCase": false
+    }
+  },
```
Comment (NikitaShkaruba) [inline src/parsing/parsers/generic/jison/delete/delete.test.json:34]: Suggest delete when your query is empty

File: src/parsing/parsers/generic/jison/delete/delete.test.json [hunk 6]
```diff
@@ -19,5 +19,127 @@
...
+    "beforeCursor": "DELETE ",
+    "afterCursor": "",
+    "containsKeywords": [
+      "FROM"
+    ],
+    "noErrors": true,
+    "expectedResult": {
+      "lowerCase": false
+    }
+  },
```
Comment (NikitaShkaruba) [inline src/parsing/parsers/generic/jison/delete/delete.test.json:46]: Suggest FROM when your query is `DELETE |`

File: src/parsing/parsers/generic/jison/delete/delete.test.json [hunk 7]
```diff
@@ -19,5 +19,127 @@
...
+    "afterCursor": "",
+    "noErrors": true,
+    "expectedResult": {
+      "lowerCase": false,
+      "suggestTables": {},
+      "suggestDatabases": {
+        "appendDot": true
+      }
+    }
+  },
```
Comment (NikitaShkaruba) [inline src/parsing/parsers/generic/jison/delete/delete.test.json:59]: Suggest tables when your query is `DELETE FROM |`

File: src/parsing/parsers/generic/jison/delete/delete.test.json [hunk 8]
```diff
@@ -19,5 +19,127 @@
...
+          },
+          "identifierChain": [
+            {
+              "name": "mytable"
+            }
+          ]
+        }
+      ]
+    }
+  },
```
Comment (NikitaShkaruba) [inline src/parsing/parsers/generic/jison/delete/delete.test.json:95]: Suggest `WHERE` when your query is `DELETE FROM mytable`.

Locations in expectedResult are the tokens our query is composed from. In our case it's delete statement, and the table. `parser.addTablePrimary($3);` in `delete.jison` adds it

File: src/parsing/parsers/generic/jison/delete/delete.test.json [hunk 9]
```diff
@@ -19,5 +19,127 @@
...
+            }
+          ]
+        }
+      ],
+      "suggestFunctions": {},
+      "suggestKeywords": [
+        "EXISTS",
+        "NOT EXISTS"
+      ]
+    }
```
Comment (NikitaShkaruba) [inline src/parsing/parsers/generic/jison/delete/delete.test.json:143]: Suggest columns, functions and keywords when your query is `DELETE FROM mytable WHERE |`

File: package.json [hunk 10]
```diff
@@ -23,8 +23,8 @@
   },
   "scripts": {
     "postinstall": "cd src/generator && npm install",
-    "pretest": "npm run generate",
-    "test": "jest src/parsing",
+    "test": "npm run generate && npm run test_without_generate",
+    "test_without_generate": "jest src/parsing",
```
Comment (NikitaShkaruba) [inline package.json:27]: Sometimes you don't want to generate a parser and just want to run tests

**Review by NikitaShkaruba — COMMENTED** [review]
Some help for the reviewer

**Review by viladimiru — APPROVED** [review]
Thanks for explanation comments!
Idk how parser inherits DataManipulation_EDIT, hope you will explain me on meeting.
Imo it should work!)

**Pull request merged**